File: experiment-15-sonnet5/brazil/runs/language=typescript_model=sonnet-5_prompt=tdd/rep1/tests/dates.test.ts
import { describe, it, expect } from "vitest";
import { parseFlexibleDate, formatISODate, isWithinDateRange } from "../src/dates.js";

describe("parseFlexibleDate", () => {
  it("parses ISO date-only strings", () => {
    const d = parseFlexibleDate("2023-09-24");
    expect(d.getUTCFullYear()).toBe(2023);
    expect(d.getUTCMonth()).toBe(8);
    expect(d.getUTCDate()).toBe(24);
  });

  it("parses ISO datetime strings with time", () => {
    const d = parseFlexibleDate("2012-05-19 18:30:00");
    expect(d.getUTCFullYear()).toBe(2012);
    expect(d.getUTCMonth()).toBe(4);
    expect(d.getUTCDate()).toBe(19);
    expect(d.getUTCHours()).toBe(18);
    expect(d.getUTCMinutes()).toBe(30);
  });

  it("parses Brazilian DD/MM/YYYY strings", () => {
    const d = parseFlexibleDate("29/03/2003");
    expect(d.getUTCFullYear()).toBe(2003);
    expect(d.getUTCMonth()).toBe(2);
    expect(d.getUTCDate()).toBe(29);
  });

  it("throws on an unparseable date", () => {
    expect(() => parseFlexibleDate("not-a-date")).toThrow();
  });
});

describe("formatISODate", () => {
  it("formats a parsed ISO datetime back to YYYY-MM-DD", () => {
    expect(formatISODate(parseFlexibleDate("2012-05-19 18:30:00"))).toBe("2012-05-19");
  });

  it("formats a parsed Brazilian date to YYYY-MM-DD", () => {
    expect(formatISODate(parseFlexibleDate("29/03/2003"))).toBe("2003-03-29");
  });
});

describe("isWithinDateRange", () => {
  it("returns true when the date falls within an inclusive range", () => {
    const d = parseFlexibleDate("2020-06-15");
    expect(isWithinDateRange(d, "2020-01-01", "2020-12-31")).toBe(true);
  });

  it("returns false when the date falls outside the range", () => {
    const d = parseFlexibleDate("2019-06-15");
    expect(isWithinDateRange(d, "2020-01-01", "2020-12-31")).toBe(false);
  });

  it("treats missing bounds as open-ended", () => {
    const d = parseFlexibleDate("2020-06-15");
    expect(isWithinDateRange(d, undefined, undefined)).toBe(true);
    expect(isWithinDateRange(d, "2020-06-15", undefined)).toBe(true);
    expect(isWithinDateRange(d, undefined, "2020-06-15")).toBe(true);
  });
});

File: experiments/adrianco/experiment-14/runs/language=clojure_model=sonnet_prompt=TDD/rep1/brazilian-soccer-mcp/test/brazilian_soccer_mcp/dates_test.clj
(ns brazilian-soccer-mcp.dates-test
  (:require [clojure.test :refer :all]
            [brazilian-soccer-mcp.dates :as dates]))

(deftest parse-date-test
  (testing "parses ISO date format"
    (let [d (dates/parse-date "2023-09-24")]
      (is (not (nil? d)))
      (is (= 2023 (dates/year d)))
      (is (= 9 (dates/month d)))
      (is (= 24 (dates/day d)))))

  (testing "parses Brazilian date format DD/MM/YYYY"
    (let [d (dates/parse-date "29/03/2003")]
      (is (not (nil? d)))
      (is (= 2003 (dates/year d)))
      (is (= 3 (dates/month d)))
      (is (= 29 (dates/day d)))))

  (testing "parses datetime format with time"
    (let [d (dates/parse-date "2012-05-19 18:30:00")]
      (is (not (nil? d)))
      (is (= 2012 (dates/year d)))
      (is (= 5 (dates/month d)))
      (is (= 19 (dates/day d)))))

  (testing "returns nil for nil or empty input"
    (is (nil? (dates/parse-date nil)))
    (is (nil? (dates/parse-date ""))))

  (testing "returns nil for unparseable date"
    (is (nil? (dates/parse-date "not-a-date")))))

(deftest date-in-range-test
  (testing "date within range is true"
    (let [d (dates/parse-date "2023-06-15")]
      (is (dates/date-in-range? d "2023-01-01" "2023-12-31"))))

  (testing "date outside range is false"
    (let [d (dates/parse-date "2022-06-15")]
      (is (not (dates/date-in-range? d "2023-01-01" "2023-12-31")))))

  (testing "nil bounds means no constraint"
    (let [d (dates/parse-date "2023-06-15")]
      (is (dates/date-in-range? d nil nil))
      (is (dates/date-in-range? d "2023-01-01" nil))
      (is (dates/date-in-range? d nil "2023-12-31"))))

  (testing "nil date returns false"
    (is (not (dates/date-in-range? nil "2023-01-01" "2023-12-31")))))

(deftest date-year-test
  (testing "extracts year from various formats"
    (is (= 2023 (dates/extract-year "2023-09-24")))
    (is (= 2003 (dates/extract-year "29/03/2003")))
    (is (= 2012 (dates/extract-year "2012-05-19 18:30:00")))
    (is (nil? (dates/extract-year nil)))
    (is (nil? (dates/extract-year "")))))
